sources/Launcher/DYNDataInterfaceContainer.cpp

```diff
@@ -76,24 +79,24 @@ class DataInterfaceContainer::DataInterfaceContainerImpl {
 
 class DataInterfaceContainer::DataInterfaceContainerImpl {
  public:
-  explicit DataInterfaceContainerImpl(const boost::shared_ptr<DYN::DataInterface>& data) : referenceDataInterface_(data) {}
+  explicit DataInterfaceContainerImpl(const std::string& iidmFile, unsigned int) : iidmFile_(iidmFile) {}
 
   void initDataInterface(unsigned int) {
-    dataInterface_ = referenceDataInterface_->clone();
+    // do nothing
   }
 
   boost::shared_ptr<DYN::DataInterface> getDataInterface() const {
-    return dataInterface_;
+    // re-read the IIDM file every time
+    return DYN::DataInterfaceFactory::build(DYN::DataInterfaceFactory::DATAINTERFACE_IIDM, iidmFile_);
```

Comment:
This means that if we call twice getDataInterface we will have 2 different Data interfaces. It would be cleaner to have the same behavior as the powsybl case: read in initDataInterface and then always return the proper Data Interface

Reply:
We cannot do that since we cannot use the clone in case powsybl is not used

Reply:
And that means we'll have to introduce a mutex in legacy where it is not needed

Reply:
well this is a dangerous behavior so we need to figure out a solution

Reply:
The point of the class is to hide the sequence to build the data interface that will be use. Changing the sequence between with and without powsybl is the point of the class. I can add in comments that it is not guaranteed that the data interface is the same if you call the API twice

Reply:
This is very error prone to have a different behavior depending on the library used, and anyway I would like to avoid the mutex in the powsybl version so I guess we will have to think about a different solution, for example not creating the variants in the build in data interface but later to be able to clone the data interfaces outside of the MT environment

Reply:
let's discuss about this at some point

Reply:
This will be solved in another issue